### PR TITLE
EVM Staking - Adds wallet balance & total staked info container

### LINKED ISSUE #1763: [TASK] Implement wallet balance and staking info container
## Description
Develop and implement container component that displays information related to the user's TNT wallet balance, rewarded TNT (total payouts), staked TNT, and bonded TNT. 

**Available TNT in Wallet & Total Rewarded TNT:**
- "Available TNT" represents what the user has at their disposal, while "Rewarded TNT' represents the user's total payouts from staking activity. 

**Total Staked TNT & Unbonding TNT:**
- These values pertain to funds that are "locked" or committed in some way. The "Staked TNT" represents the user's active investment in the network, and the "Unbonding TNT" implies funds that are in a transitional phase but has not yet been released to user's wallet due to the unbonding period.

Please reference to Figma wireframes for specific implementation (i.e., wallet disconnected view). 

### Screenshots
Screenshot 2023-10-11 at 2 42 43 PM

### PR DESCRIPTION
**Pull request by devpavan04**
## Summary of changes

- Upon connecting to wallet, nominators can now see their `TNT` balance in their wallet.
- Also, post connecting they can see how much total stake they have in the network.

### Proposed area of change

- [ ] `apps/bridge-dapp`
- [ ] `apps/hubble-stats`
- [ ] `apps/stats-dapp`
- [x] `apps/tangle-dapp`
- [ ] `apps/faucet`
- [ ] `libs/webb-ui-components`

### Reference issue to close (if applicable)

- Closes #1763 

### Screen Recording

https://github.com/webb-tools/webb-dapp/assets/53374218/59a89681-1147-45d9-a14d-5a97e831da89

#### Responsiveness

https://github.com/webb-tools/webb-dapp/assets/53374218/4f5cec98-54c6-4faa-a7cb-a4f4554f21c2

### CODE REVIEW

File: apps/tangle-dapp/components/StatsMetricItem/types.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { StatsMetricReturnType } from '../../types';
+
+export interface StatsMetricItemProps {
+  title: string;
+  tooltip?: string;
+  dataFetcher: (
+    address: string | `0x${string}`
+  ) => Promise<StatsMetricReturnType>;
+  address: string | `0x${string}`;
```
Comment (vutuanlinh2k2) [inline apps/tangle-dapp/components/StatsMetricItem/types.ts:9]: ``0x${string}`` seems to be redundant here

File: apps/tangle-dapp/components/StatsMetricItem/StatsMetricItem.tsx
Comment (vutuanlinh2k2) [inline apps/tangle-dapp/components/StatsMetricItem/StatsMetricItem.tsx:1]: minor improvements to follow to have better code convention: add `type FC` in line 2 and comment `/** @internal */` before line 34

File: apps/tangle-dapp/utils/splitTokenValueAndSymbol.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ *
+ * This function splits a string into a numeric (Token value) and alphabets (Token symbol) part.
+ * @param string - The string to split
+ * @returns - An object containing the numeric and alphabets part of the string
+ */
+export const splitTokenValueAndSymbol = (
+  string: string
+): SplitNumericAndAlphabetsReturnType => {
+  const regex = /^(\d+\.?\d*)\s*(\D+)$/;
```
Comment (vutuanlinh2k2) [inline apps/tangle-dapp/utils/splitTokenValueAndSymbol.ts:15]: add comment here to explain the regex

**Review by vutuanlinh2k2 — CHANGES_REQUESTED** [review]
left some comments

File: apps/tangle-dapp/utils/splitTokenValueAndSymbol.ts (hunk 2 above)
Comment (AtelyPham) [reply apps/tangle-dapp/utils/splitTokenValueAndSymbol.ts:19]: Let's use `/** */` for multiple lines comments. That also supports for the documenting the variable. 

CleanShot 2023-11-22 at 18 10 40

**Review by AtelyPham — COMMENTED** [review]
Left some comments.

File: apps/tangle-dapp/constants/evm.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      webSocket: ['wss://rpc-archive.tangle.tools'],
+    },
+  },
+  blockExplorers: {
+    default: {
+      name: 'Tangle Testnet Explorer',
+      url: 'https://explorer.tangle.tools',
+    },
+  },
+});
```
[the remark this answers is not in the document]
Comment (devpavan04) [reply apps/tangle-dapp/constants/evm.ts]: In the `chainsConfig`, the `TangleTestnet` includes a `contracts` property that is not required in `tangle-dapp`. Thanks for pointing this out @AtelyPham, I have now adopted the same configuration but removed the contracts property.

**Pull request merged**